bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java

```diff
@@ -282,6 +279,93 @@ static int callAndWait(String[] pstr, ToIntFunction<IUnknown> callable) {
 	return phr[0];
 }
 
+class WebViewProvider {
+
+	private CompletableFuture<ICoreWebView2> webViewFuture = new CompletableFuture<>();
+	private CompletableFuture<ICoreWebView2_2> webView_2Future = new CompletableFuture<>();
+	private CompletableFuture<ICoreWebView2_11> webView_11Future = new CompletableFuture<>();
+
+	private CompletableFuture<Void> lastWebViewTask = webViewFuture.thenRun(() -> {});
+
+	ICoreWebView2 initializeWebView(ICoreWebView2Controller controller) {
+		long[] ppv = new long[1];
+		controller.get_CoreWebView2(ppv);
+		final ICoreWebView2 webView = new ICoreWebView2(ppv[0]);
+		initializeWebView_2(webView);
+		initializeWebView_11(webView);
+		webViewFuture.complete(webView);
+		return webView;
+	}
+
+	private void initializeWebView_2(ICoreWebView2 webView) {
+		long[] ppv = new long[1];
+		int hr = webView.QueryInterface(COM.IID_ICoreWebView2_2, ppv);
+		if (hr == COM.S_OK) {
+			webView_2Future.complete(new ICoreWebView2_2(ppv[0]));
+		} else {
+			webView_2Future.cancel(true);
+		}
+	}
+
+	private void initializeWebView_11(ICoreWebView2 webView) {
+		long[] ppv = new long[1];
+		int hr = webView.QueryInterface(COM.IID_ICoreWebView2_11, ppv);
+		if (hr == COM.S_OK) {
+			webView_11Future.complete(new ICoreWebView2_11(ppv[0]));
+		} else {
+			webView_11Future.cancel(true);
+		}
+	}
+
+	ICoreWebView2 getWebView(boolean waitForPendingWebviewTasksToFinish) {
```

Comment:
```suggestion
	private ICoreWebView2 getWebView(boolean waitForPendingWebviewTasksToFinish) {
```

Reply:
btw getWebView is used by the Edge class. Wasnt this agreed that for all the methods which is not used outside the WebViewProvider, we use private and all the methods used by Edge we use package protected?

Reply:
Yes, this is misplaced. Sorry for that. I think I wanted to mark some other method, but all visibilities seem to be fine for all of them now.